### PR TITLE
MVP saf-cli Lambda Function

### PR DESCRIPTION
**Pull request by aaronlippold**
MVP of the `saf-cli` lambda function. The function will supporting reading HDF data out of an s3 bucket and providing those file(s) to the SAF CLI app and then either pushing the result to the saf convert functions end-point and or writting the result to the perscribed s3 target bucket.

Items Left To Resolve
- [ ] Properly param the input and outputs of the function, right now we are hardcoded to hdf2splunk
- [ ] Remove any hard-coded data from above
- [ ] Update the README and document installing the function
- [ ] add unit-tests and mocks 
- [ ] add unit-test and mock data 
- [ ] Document local / mock testing
- [ ] Document installation and update process
- [ ] Break any repeated code into seperate file modules for import
- [ ] Decide if we Migrate to the serverless framework to get some of the above
- [ ] Publish on the `AWS Marketplace of Functions` - find out what that means :)
- [ ] have a 🍺

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    logger.info(view_heimdall_message);
+    throw new Error(view_heimdall_message);
+  }
+
+  // TODO: REMOVE ALL THESE COMMENTS
+  // TODO: Decide is we want to catch undefined saf-cli command groupings https://stackoverflow.com/questions/15201939/jquery-javascript-check-string-for-multiple-substringsa
+  // TODO: Removed hardcoded data and move to lambda params
+
+  const HEC_TOKEN = "473b3297-1d88-4740-96ff-e6048e51b785";
+  const SPLUNK_SERVER = "splk1.efficacy.online";
```
Comment (camdenmoors) [inline app.js:63]: Might want to remove this and clear git history
```suggestion
  const HEC_TOKEN = "YOUR_HEC_TOKEN";
  const SPLUNK_SERVER = "YOUR_SPLUNK_SERVER";
```

File: app.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    logger.info("Reading File:     " + params.Key);
+
+    let { ContentType, Body } = await s3.getObject(params).promise();
+
+    logger.debug("Recieved File ContentType - " + ContentType);
+
+    let HDF_FILE = path.resolve("/tmp/", params.Key.toString());
+    Body = Body.toString();
+
+    const command_string = [CLI_COMMAND + ":" + CLI_FUNCTION, "-i", HDF_FILE, "-H", SPLUNK_SERVER, "-t", HEC_TOKEN];
```
Comment (camdenmoors) [inline app.js:100]: The HDF2Splunk mapper is going to be taking a username/password or token in the future.

Splunk options: https://github.com/mitre/heimdall2/pull/2675/commits/d183b12bfa9f183de9f4b6e3a671346740e72d62#diff-19410c8789053ca0d35bbdd8acd403aa1e4165ed4b17d9733d763242817e9694R21-R26

File: app.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  if (CLI_COMMAND.trim() === "view" && CLI_FUNCTION.trim() === "heimdall") {
+    let view_heimdall_message = "You cannot use the 'saf view:heimdall' command in this environment.";
+    logger.info(view_heimdall_message);
+    throw new Error(view_heimdall_message);
+  }
+
+  // TODO: REMOVE ALL THESE COMMENTS
+  // TODO: Decide is we want to catch undefined saf-cli command groupings https://stackoverflow.com/questions/15201939/jquery-javascript-check-string-for-multiple-substringsa
```
Comment (em-c-rod) [inline app.js:59]: I added this check in `saf_action` if that helps.
https://github.com/mitre/saf_action/blob/main/index.js#L9

**Comment by aaronlippold** [comment]
Closing in favor of #3 

Please move your comments and reviews there

**Pull request closed without merging**